bin/consequence_mapping/consequence_mapping.py

```diff
@@ -0,0 +1,154 @@
+        if len(list_of_consequences) == 0
+    })
+
+
+def process_variants(variants):
+    """Given a list of variant IDs, return a list of consequence types (each including Ensembl gene name & ID and a
+    functional consequence code) for a given variant."""
+
+    # First, we query VEP with default parameters, looking for variants affecting protein coding and miRNA transcripts
+    # up to a standard distance (5000 nucleotides either way, which is default for VEP) from the variant.
+    results_by_variant = {}
+    vep_results = query_vep(variants=variants, search_distance=5000)
+    extract_consequences(vep_results=vep_results, acceptable_biotypes={'protein_coding', 'miRNA'},
+                         only_closest=False, results_by_variant=results_by_variant)
+
+    # See if there are variants with no consequences up to the default distance
+    variants_without_consequences = get_variants_without_consequences(results_by_variant)
+    # If there are, we will now do a second round of querying, this time looking only at protein coding biotypes (vs.
+    # miRNA *and* protein coding during the first round) up to a distance of 500,000 bases each way.
+    if variants_without_consequences:
+        logger.info('Found {} variant(s) without standard consequences: {}. Querying distant regions'.format(
+            len(variants_without_consequences), '|'.join(variants_without_consequences)))
+        distant_vep_results = query_vep(variants=variants_without_consequences, search_distance=500000)
+        extract_consequences(vep_results=distant_vep_results, acceptable_biotypes={'protein_coding'},
+                             only_closest=True, results_by_variant=results_by_variant)
+
+    # See if there are still variants with no consequences, even up to a wide search window
+    variants_without_consequences = get_variants_without_consequences(results_by_variant)
+    if variants_without_consequences:
+        logger.info('After distant querying, still remaining {} variant(s) without consequences: {}'.format(
+            len(variants_without_consequences), '|'.join(variants_without_consequences)
+        ))
+
+    # Yield all consequences for all variants. Note they are not grouped by variant, all consequences are yielded in a
+    # common sequence.
+    for variant_id, variant_consequences in results_by_variant.items():
+        for consequence_to_yield in deduplicate_list(variant_consequences):
+            yield consequence_to_yield
+
+
+if __name__ == '__main__':
+
+    # Load severity rankings for consequence terms
+    severity_ranking_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'severity_ranking.txt')
+    severity_ranking = open(severity_ranking_path).read().splitlines()
+    consequence_term_severity_rank = {term: index for index, term in enumerate(severity_ranking)}
+
+    variants_to_query = [colon_to_vep(v) for v in sys.stdin.read().splitlines()]
```

Comment:
Would suggest making it explicit in a comment that this program expects data from a pipe...

Reply:
Done